components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Publisher/src/Deploy/IoTHubPublisherDeployment.cs

```diff
@@ -93,7 +93,7 @@ public sealed class IoTHubPublisherDeployment : IHostProcess {
             {
                 ""$edgeAgent"": {
                     " + registryCredentials + @"
-                    ""properties.desired.modules.opcpublisher"": {
+                    ""properties.desired.modules.publisher"": {
```

Comment:
Because on upgrade old identity is not deleted we have dead identities flying around then.   If this were important then we can open a work item, cost it and decide whether the cost is worth the change.